src/mon/FSCommands.cc

```diff
@@ -1261,6 +1261,172 @@ class RenameFilesystemHandler : public FileSystemCommandHandler
+
+    // Check that FSCID provided for both CephFSs is correct.
+    if (fs1_id != fs1p->get_fscid() || fs2_id != fs2p->get_fscid()) {
+      if (fs1_id != fs1p->get_fscid() && fs2_id == fs2p->get_fscid()) {
+	ss << "FSCID provided for \"" << fs1_name << "\" is incorrect.";
+	return -EINVAL;
+      } else if (fs1_id == fs1p->get_fscid() && fs2_id != fs2p->get_fscid()) {
+	ss << "FSCID provided for \"" << fs2_name << "\" is incorrect.";
+	return -EINVAL;
+      } else if (fs1_id != fs1p->get_fscid() && fs2_id != fs2p->get_fscid()) {
+	ss << "FSCIDs provided for both the CephFSs is incorrect.";
+	return -EINVAL;
+      }
+    }
+
+    // Check that CephFS mirroring for both CephFSs is disabled.
+    if (fs1p->get_mirror_info().mirrored || fs2p->get_mirror_info().mirrored) {
+      if (fs1p->get_mirror_info().mirrored &&
+	  !fs2p->get_mirror_info().mirrored) {
+	ss << "Mirroring is enabled on file system \""<< fs1_name << "\". "
+	   << "Disable mirroring on the file system after ensuring it's OK "
+	   << "to do so, and then re-try swapping.";
+	return -EPERM;
+      } else if (!fs1p->get_mirror_info().mirrored &&
+		 fs2p->get_mirror_info().mirrored) {
+	ss << "Mirroring is enabled on file system \""<< fs2_name << "\". "
+	   << "Disable mirroring on the file system after ensuring it's OK "
+	   << "to do so, and then re-try swapping.";
+	return -EPERM;
+      } else {
+	ss << "Mirroring is enabled on file systems \"" << fs1_name << "\" "
+	   << "and \"" << fs2_name << "\". Disable mirroring on both the "
+	   << "file systems after ensuring it's OK to do so, and then re-try "
+	   << "swapping.";
+	return -EPERM;
+      }
+    }
+
+    if (!mon->osdmon()->is_writeable()) {
+      // not allowed to write yet, so retry when we can
+      mon->osdmon()->wait_for_writeable(
+	op, new PaxosService::C_RetryMessage(mon->mdsmon(), op));
+      return -EAGAIN;
+    }
+
+    // Check that both CephFS have been marked as down, IOW has no MDS
+    // associated with it.
+    if (fs1p->get_mds_map().get_num_up_mds() > 0 ||
+        fs2p->get_mds_map().get_num_up_mds() > 0) {
+      if (fs1p->get_mds_map().get_num_up_mds() > 0 &&
+          fs2p->get_mds_map().get_num_up_mds() == 0) {
+	ss << "CephFS \"" << fs1_name << "\" has not been failed. Before "
+	   << "swapping CephFS names, both CephFSs should be marked as "
+	   << "failed. See `ceph fs fail`.";
+	return -EINVAL;
+      } else if (fs1p->get_mds_map().get_num_up_mds() == 0 &&
+		 fs2p->get_mds_map().get_num_up_mds() > 0) {
+	ss << "CephFS \"" << fs2_name << "\" has not been failed. Before "
+	   << "swapping CephFS names, both CephFSs should be marked as "
+	   << "failed. See `ceph fs fail`.";
+	return -EINVAL;
+      } else {
+	ss << "CephFSs \"" << fs1_name << "\" and \"" << fs2_name << "\" has "
+	   << "not been failed. Before swapping CephFS names, both CephFSs "
+	   << "should be marked as failed. See `ceph fs fail`.";
+	return -EINVAL;
+      }
+    }
+
+    // Finally, the swap begins.
```

Comment:
So, @batrick mentioned in slack about setting `refuse_client_session` and something about wiping the sessions. I'm not totally clear about the intent. Could you explain @rishabh-d-dave ?

Reply:
I don't know anything besides what he posted on Slack channel for CephFS. He was basically looking forward to make it difficult for user to shoot itself in the foot while using this command.

Reply:
I spoke regarding this in standup yesterday - the idea is to set `refuse_client_session` during the swap operation. When the MDS join the cluster again, client will not be allowed to establish a session with the MDS. The cluster admin would have to manually unset this config and  remount the clients.

Reply:
Like `fs fail` won't it be better if user has to do it manually?

Reply:
In that case the operation should not proceed if the relevant flag isn't enabled.

Reply:
Okay. I'll add a check to make sure that `fs swap` command fails if `refuse_client_session` is not set.

Reply:
Done.